### PR TITLE
Feature/initial files

### PR DESCRIPTION
**Pull request by CBarrett2**
Added gruntfile with default and watch tasks to check the coding style using JShint and JSCS. Also added example files into src, which create a simple chart.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "validthis": true,
+
+    "browser": true,
+    "devel": true,
+    "jasmine": true,
+
+    "globals": {
+        "d3": false,
+        "fc": false,
+        "require": false
```
Comment (jleft) [inline .jshintrc]: We're not using `require`, so we can remove this for now.

File: Gruntfile.js [hunk 2]
```diff
@@ -1,8 +1,76 @@
+/* global module */
+
 module.exports = function(grunt) {
+    'use strict';
+    grunt.initConfig({
+        pkg: grunt.file.readJSON('package.json'),
+
+        meta: {
+            componentsJsFiles: [
```
Comment (jleft) [inline Gruntfile.js]: I think `componentsJsFiles` is a bit misleading, as we're build an app rather than components, maybe `srcJsFiles`?

File: Gruntfile.js [hunk 3]
```diff
@@ -1,8 +1,76 @@
...
+                    src: ['<%= meta.ourJsFiles %>']
+                }
+            }
+        },
 
-  grunt.initConfig({
-  });
+        watch: {
+            files: ['<%= meta.ourJsFiles %>'],
+            tasks: ['jshint', 'jscs']
```
Comment (jleft) [inline Gruntfile.js]: Can we just call `default` here?

File: Gruntfile.js [hunk 4]
```diff
@@ -1,8 +1,76 @@
...
-  grunt.registerTask('default', []);
+    grunt.loadNpmTasks('grunt-contrib-jshint');
+    grunt.loadNpmTasks('grunt-jscs');
+    grunt.loadNpmTasks('grunt-contrib-watch');
 
+    grunt.registerTask('default', ['jshint', 'jscs']);
+    grunt.registerTask('check:failOnError', ['jshint:failOnError', 'jscs:failOnError']);
+    grunt.registerTask('check:warnOnly', ['jshint:warnOnly', 'jscs:warnOnly']);
+    grunt.registerTask('check', ['check:failOnError']);
+    grunt.registerTask('check:failOnError', ['jshint:failOnError', 'jscs:failOnError']);
```
Comment (jleft) [inline Gruntfile.js]: `check:warnOnly` and `check:failOnError` have been included twice.

File: src/index.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE HTML>
+<html lang="en">
+	<div class="container-fluid">
```
Comment (jleft) [inline src/index.html]: The `head` element shouldn't be surrounded by this `div`. `head` should be first element in the `html`, it only contains meta data about the document, so doesn't need a `div` to position/style it etc. The `head` doesn't need a class either.

http://www.w3.org/TR/html5/document-metadata.html#the-head-element

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+		<head class="page-header">
+			<meta charset="UTF-8">   
+			<meta http-equiv="X-UA-Compatible" content="IE=edge"> <!--Needed for bootstrap-->
+			<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no"> <!--Needed for bootstrap-->
+			<meta name="description" content="description">
+			
+			<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script> <!-- JQuery - also necessary for bootstrap -->
```
Comment (jleft) [inline src/index.html]: Not a big deal, but I think we're safe to remove jQuery, Bootstrap, html5shiv and respond for now? Just to make it a little less cluttered.

File: src/index.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+<html lang="en">
+	<div class="container-fluid">
+		<head class="page-header">
+			<meta charset="UTF-8">   
+			<meta http-equiv="X-UA-Compatible" content="IE=edge"> <!--Needed for bootstrap-->
+			<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no"> <!--Needed for bootstrap-->
+			<meta name="description" content="description">
+			
+			<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script> <!-- JQuery - also necessary for bootstrap --> 
+			<script src="bootstrap.min.js"></script> <!-- Bootstrap -->
```
Comment (jleft) [inline src/index.html]: This file doesn't exist, so I'd remove the tag.

File: src/index.html [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+			<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script> <!--These two for ie8 support of bootstrap-->
+			<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
+			<script src="http://d3js.org/d3.v3.min.js" charset="utf-8"></script> <!-- D3! -->
```
Comment (jleft) [inline src/index.html]: We don't need to include D3 twice - we can just link to the `node_modules`.

File: src/index.html [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+			<script src="node_modules/d3fc/node_modules/d3/d3.js"></script> <!-- These 3 for D3FC -->
```
Comment (jleft) [inline src/index.html]: Fix the link to the `node_modules` as they are up a level. `src="../node_modules/etc`

File: src/main.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+/* A test function */
+
+// Main function
```
Comment (jleft) [inline src/main.js]: Could strip out the comments but very minor point.

File: src/main.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+(function() {
```
Comment (jleft) [inline src/main.js]: For consistency: `(function(d3, fc) {`

File: src/main.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        .attr('width', width)
+        .attr('height', height);
+
+    var data = fc.dataGenerator()(150);
+
+    var chart = fc.chart.linearTimeSeries()
+        .xDomain(fc.util.extent(data, 'date'))
+        .yDomain(fc.util.extent(data, ['open', 'close']));
+
+    svg.datum(data)
```
Comment (jleft) [inline src/main.js]: I think this:

```js
svg.datum(data)
    .call(chart);
```

should be called last, otherwise we won't see the multi series.

File: src/main.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+    var gridlines = fc.annotation.gridline()
+        .yTicks(5)
+        .xTicks(0);
+
+    var multi = fc.series.multi()
+        .series([gridlines, area, line]);
+
+    chart.plotArea(multi);
+})();
```
Comment (jleft) [inline src/main.js]: For consistency: `})(d3, fc);`

File: src/style.css [hunk 14]
```diff
@@ -0,0 +1,14 @@
+path.area {
+  fill: url(#area-gradient);
```
Comment (jleft) [inline src/style.css]: I don't think `#area-gradient` is defined? We could just set a solid fill or use the default and remove this class?

**Comment by jleft** [comment]
Thanks for this @CBarrett2, this looks like a really good start to me, just a few minor comments, but I think it'll be good to merge once the amends have been made!

**Comment by CBarrett2** [comment]
Fixed everything you suggested :)

File: Gruntfile.js [hunk 15]
```diff
@@ -1,8 +1,74 @@
...
+            tasks: ['check']
+        }
+    });
 
-  grunt.registerTask('default', []);
+    grunt.loadNpmTasks('grunt-contrib-jshint');
+    grunt.loadNpmTasks('grunt-jscs');
+    grunt.loadNpmTasks('grunt-contrib-watch');
 
+    grunt.registerTask('default', ['jshint', 'jscs']);
```
Comment (jleft) [inline Gruntfile.js]: Shall we change `['jshint', 'jscs']` to `['check']`, then we won't be running both `failOnError` and `warnOnly` for both tasks?

File: src/index.html [hunk 16]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE HTML>
```
Comment (jleft) [inline src/index.html]: This file has mixed tabs and spaces - shall we go with 2 spaces for now?

File: src/style.css [hunk 17]
```diff
@@ -0,0 +1,9 @@
+path.line {
+  stroke: rgb(26, 154, 249);
+}
+
+#chart-example {
+    position: absolute;
```
Comment (jleft) [inline src/style.css]: I think 2 space indentation would be good here too.

**Comment by jleft** [comment]
> Fixed everything you suggested :)

Brilliant, thank you!

Just a couple more pedantic comments, but then this should be good to go.

**Comment by CBarrett2** [comment]
Fixed :D

**Comment by jleft** [comment]
> Fixed :D

Thanks :smiley:
